File: Scripts/results/generated_code_results/Flutter/News&Information/CNN/settingsScreen.dart
import 'package:flutter/material.dart';
class SettingsScreen extends StatefulWidget {
@override
_SettingsScreenState createState() => _SettingsScreenState();
}
class _SettingsScreenState extends State<SettingsScreen> {
bool _enableThisIsCNN = false;
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
title: Text('Settings'),
actions: [
Icon(Icons.sd_card),
SizedBox(width: 10),
Icon(Icons.battery_full),
],
),
body: ListView(
children: [
_buildSectionHeader('ACCOUNT'),
ListTile(
title: Text('Log In to your CNN Account'),
onTap: () => Navigator.pushNamed(context, 'LogIn'),
),
Divider(),
ListTile(
title: Text('Create a free CNN Account'),
onTap: () => Navigator.pushNamed(context, 'CreateAccont'),
),
Divider(),
ListTile(
title: Text('Saved Stories'),
subtitle: Text('Find your saved articles, videos & audio here'),
onTap: () {
// Navigate to Saved Stories
},
),
_buildSectionHeader('APP PREFERENCES'),
ListTile(
title: Text('Edition'),
subtitle: Text('International'),
onTap: () {
// Open Edition selection
},
),
Divider(),
ListTile(
title: Text('Alerts'),
onTap: () {
// Open Alerts settings
},
),
Divider(),
ListTile(
title: Text('Android Notification Settings'),
onTap: () {
// Open Android Notification Settings
},
),
Divider(),
SwitchListTile(
title: Text('Enable 'This is CNN''),
subtitle: Text('A sound effect when you open the app'),
value: _enableThisIsCNN,
onChanged: (bool value) {
setState(() {
_enableThisIsCNN = value;
});
},
),
Divider(),
Padding(
padding: const EdgeInsets.all(16.0),
child: Text(
'You are currently viewing CNN in light mode. To switch to dark mode, adjust the display setting on your device.',
style: TextStyle(color: Colors.grey),
),
),
_buildSectionHeader('GENERAL'),
ListTile(
title: Text('Send Feedback'),
onTap: () {
// Open feedback form
},
),
],
),
bottomNavigationBar: BottomNavigationBar(
currentIndex: 3,
type: BottomNavigationBarType.fixed,
items: [
BottomNavigationBarItem(icon: Icon(Icons.home), label: 'Home'),
BottomNavigationBarItem(icon: Icon(Icons.how_to_vote), label: 'Elections'),
BottomNavigationBarItem(icon: Icon(Icons.play_circle_outline), label: 'Watch'),
BottomNavigationBarItem(icon: Icon(Icons.person), label: 'Settings'),
],
onTap: (index) {
switch (index) {
case 0:
Navigator.pushNamed(context, 'Home');
break;
case 1:
Navigator.pushNamed(context, 'Elections');
break;
case 2:
// Handle Watch navigation
break;
}
},
),
);
}
Widget _buildSectionHeader(String title) {
return Padding(
padding: const EdgeInsets.fromLTRB(16, 24, 16, 8),
child: Text(
title,
style: TextStyle(
fontSize: 14,
fontWeight: FontWeight.bold,
color: Colors.grey[600],
),
),
);
}
}

File: Scripts/results/generated_code_results/ablation_study/sam/News&Information/CNN/SettingsScreen.js
import React, { useState } from 'react';

import { View, Text, StyleSheet, TouchableOpacity, Switch, ScrollView } from 'react-native';

import Icon from 'react-native-vector-icons/MaterialIcons';



const SettingsScreen = ({ navigation }) => {

  const [enableThisIsCNN, setEnableThisIsCNN] = useState(false);



  const renderSettingsItem = (title, subtitle = null, onPress = null, rightElement = null) => (

    <TouchableOpacity style={styles.settingsItem} onPress={onPress}>

      <View>

        <Text style={styles.settingsItemTitle}>{title}</Text>

        {subtitle && <Text style={styles.settingsItemSubtitle}>{subtitle}</Text>}

      </View>

      {rightElement || <Icon name="chevron-right" size={24} color="#888" />}

    </TouchableOpacity>

  );



  return (

    <ScrollView style={styles.container}>

      <Text style={styles.header}>Settings</Text>



      <Text style={styles.sectionHeader}>ACCOUNT</Text>

      {renderSettingsItem('Log In to your CNN Account', null, () => navigation.navigate('LogIn'))}

      {renderSettingsItem('Create a free CNN Account', null, () => navigation.navigate('CreateAccont'))}

      {renderSettingsItem('Saved Stories', 'Find your saved articles, videos & audio here')}



      <Text style={styles.sectionHeader}>APP PREFERENCES</Text>

      {renderSettingsItem('Edition', 'International')}

      {renderSettingsItem('Alerts')}

      {renderSettingsItem('Android Notification Settings')}

      {renderSettingsItem(

        'Enable \'This is CNN\'',

        'A sound effect when you open the app',

        null,

        <Switch

          value={enableThisIsCNN}

          onValueChange={(value) => setEnableThisIsCNN(value)}

        />

      )}



      <Text style={styles.darkModeText}>

        You are currently viewing CNN in light mode. To switch to dark mode, adjust the display setting on your device.

      </Text>



      <Text style={styles.sectionHeader}>GENERAL</Text>

      {renderSettingsItem('Send Feedback')}



      <View style={styles.footer}>

        <TouchableOpacity style={styles.footerButton} onPress={() => navigation.navigate('Home')}>

          <Icon name="home" size={24} color="#888" />

          <Text style={styles.footerButtonText}>Home</Text>

        </TouchableOpacity>

        <TouchableOpacity style={styles.footerButton} onPress={() => navigation.navigate('Elections')}>

          <Icon name="how-to-vote" size={24} color="#888" />

          <Text style={styles.footerButtonText}>Elections</Text>

        </TouchableOpacity>

        <TouchableOpacity style={styles.footerButton}>

          <Icon name="ondemand-video" size={24} color="#888" />

          <Text style={styles.footerButtonText}>Watch</Text>

        </TouchableOpacity>

        <TouchableOpacity style={[styles.footerButton, styles.activeFooterButton]}>

          <Icon name="settings" size={24} color="#f00" />

          <Text style={[styles.footerButtonText, styles.activeFooterButtonText]}>Settings</Text>

        </TouchableOpacity>

      </View>

    </ScrollView>

  );

};



const styles = StyleSheet.create({

  container: {

    flex: 1,

    backgroundColor: '#fff',

  },

  header: {

    fontSize: 24,

    fontWeight: 'bold',

    padding: 16,

  },

  sectionHeader: {

    fontSize: 12,

    fontWeight: 'bold',

    color: '#888',

    paddingHorizontal: 16,

    paddingTop: 16,

    paddingBottom: 8,

  },

  settingsItem: {

    flexDirection: 'row',

    justifyContent: 'space-between',

    alignItems: 'center',

    paddingVertical: 16,

    paddingHorizontal: 16,

    borderBottomWidth: 1,

    borderBottomColor: '#eee',

  },

  settingsItemTitle: {

    fontSize: 16,

  },

  settingsItemSubtitle: {

    fontSize: 12,

    color: '#888',

    marginTop: 4,

  },

  darkModeText: {

    fontSize: 14,

    color: '#333',

    padding: 16,

  },

  footer: {

    flexDirection: 'row',

    justifyContent: 'space-around',

    borderTopWidth: 1,

    borderTopColor: '#eee',

    paddingVertical: 8,

  },

  footerButton: {

    alignItems: 'center',

  },

  footerButtonText: {

    fontSize: 12,

    color: '#888',

    marginTop: 4,

  },

  activeFooterButton: {

    color: '#f00',

  },

  activeFooterButtonText: {

    color: '#f00',

  },

});



export default SettingsScreen;